src/manage_nsfs/nc_lifecycle.js

```diff
@@ -1190,11 +1213,22 @@ class NCLifecycle {
 
         await native_fs_utils._create_path(ILM_POLICIES_TMP_DIR, this.non_gpfs_fs_context, config.BASE_MODE_CONFIG_DIR);
         await native_fs_utils._create_path(ILM_CANDIDATES_TMP_DIR, this.non_gpfs_fs_context, config.BASE_MODE_CONFIG_DIR);
-        for (const [mount_point, policy] of Object.entries(mount_point_to_policy_map)) {
-            if (policy === '') continue;
-            const ilm_policy_path = await this.write_tmp_ilm_policy(mount_point, policy);
-            await this.create_candidates_file_by_gpfs_ilm_policy(mount_point, ilm_policy_path);
-        }
+        await P.map_with_concurrency(config.NC_LIFECYCLE_GPFS_MMAPPLY_ILM_POLICY_CONCURRENCY,
+            Object.entries(mount_point_to_policy_map), async ([mount_point, policy]) => {
+            try {
+                if (policy === '') return;
+                await this._call_op_and_update_status({
+                    mount_point,
+                    op_name: TIMED_OPS.CREATE_GPFS_CANDIDATE_FILE_BY_ILM_POLICY,
+                    op_func: async () => {
+                        const ilm_policy_path = await this.write_tmp_ilm_policy(mount_point, policy);
+                        await this.create_candidates_file_by_gpfs_ilm_policy(mount_point, ilm_policy_path);
+                    }
+                });
+            } catch (err) {
+                dbg.error('create_candidates_file_by_gpfs_ilm_policy failed with error', err, err.code, err.message);
```

Comment:
wouldn't we still try to parse the file in get_candidates? do we have something to protect us there?

Reply:
if the file is not created we won't parse it, see here - https://github.com/noobaa/noobaa-core/blob/master/src/manage_nsfs/nc_lifecycle.js#L1429-L1434
the reading is wrapped with a try catch clause - 
```
 } catch (err) {
            if (err.code === 'ENOENT') {
                dbg.log2(`parse_candidates_from_gpfs_ilm_policy ilm_candidates_file_exists does not exist, no candidates to delete`);
                this._set_rule_state(bucket_json, lifecycle_rule, finished_state);
                return [];
            }
```